libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -561,6 +561,12 @@ public static String optimizeImage(Context context, String path, int maxImageWid
             // Can't read the src dimensions.
             return path;
         }
+
+        // do not optimize if original-size and 100% quality are set.
+        if (maxImageWidth == Integer.MAX_VALUE && quality == 100) {
```

Comment:
What happens if `maxImageWidth == Integer.MAX_VALUE` but `quality` is **less** than 100?